### PR TITLE
[QPPA-458][Bug Fix] Include benchmarks that were excluded during conversion

### PR DESCRIPTION
**Pull request by lamroger**

### CODE REVIEW

File: staging/benchmarks/json/001_benchmarks_updates.json [hunk 1]
```diff
@@ -3162,6 +3434,40 @@
     ]
   },
   {
+    "measureId": "54",
```
Comment (lamroger) [inline staging/benchmarks/json/001_benchmarks_updates.json]: Do we have a standard format for measureIds? Should I pad numbers with leading zeros?

What happens in this PR is that if its not recognized in the measures file, it uses the measureId in the benchmark CSV which doesn't have leading 0s

File: scripts/benchmarks/format-benchmark-record.js [hunk 2]
```diff
@@ -127,6 +133,34 @@ const formatDecileGenerator = function(record) {
   };
 };
 
+// Looks in measures-data.json for an existing measureIds and
+// walks through the original, capitalized + removed spaces,
+// and capitalized + underscored spaces versions.
+// If found, returns the measureId from the measures-data.json file.
+// If none are found, return the padded number or non-spaced version
+const formatMeasureId = (measureId) => {
+  const measureIdRemovedSpaces = measureId.replace(/\s+/g, '');
```
Comment (marimiyachi) [inline scripts/benchmarks/format-benchmark-record.js]: Are there any cases with both spaces AND underscores?

File: scripts/benchmarks/format-benchmark-record.js [hunk 3]
```diff
@@ -127,6 +133,34 @@ const formatDecileGenerator = function(record) {
...
+
+  for (const formattedMeasureID of measureIdCandidates) {
+    const measure = MEASURE_ID_TO_MEASURE_MAP[formattedMeasureID];
+    if (measure) return measure.measureId;
+  };
+
+  // If all digits, pad with zeros up to the thousands
+  // else, return a nonspaced version
+  if (measureId.match(/^\d+$/)) {
+    return ('0000' + measureId).slice(-4);
```
Comment (marimiyachi) [inline scripts/benchmarks/format-benchmark-record.js]: What's the reason for padding in this case?

Comment (lamroger) [reply scripts/benchmarks/format-benchmark-record.js]: It's the format numeric-only benchmarks are using right now. There isn't a set standard but I wanted to conform to something we have already.

Comment (lamroger) [reply scripts/benchmarks/format-benchmark-record.js]: They're actually padded to the hundredth place in `benchmarks/2017.json`. Made the change and added a test.

Comment (marimiyachi) [reply scripts/benchmarks/format-benchmark-record.js]: 👍

File: scripts/benchmarks/format-benchmark-record.js (hunk 2 above)
Comment (lamroger) [reply scripts/benchmarks/format-benchmark-record.js]: Ya, it's rare but it can happen. I'll write some regex to fuzzy match

Comment (lamroger) [reply scripts/benchmarks/format-benchmark-record.js]: Rare as it someone would have to generate it and I havent seen it before

File: test/scripts/benchmarks/format-benchmark-record-spec.js [hunk 4]
```diff
@@ -1,20 +1,43 @@
...
 // Utils
-const formatBenchmarkRecord = require('./../../../scripts/benchmarks/format-benchmark-record');
+const { formatBenchmarkRecord, formatMeasureId } = require('./../../../scripts/benchmarks/format-benchmark-record');
 
 const options = {
   benchmarkYear: 2016,
   performanceYear: 2018
 };
 
+describe('formatMeasureId', function() {
```
Comment (marimiyachi) [inline test/scripts/benchmarks/format-benchmark-record-spec.js]: nice tests :)

File: test/scripts/benchmarks/parse-benchmark-data-spec.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+// Function which executes script and reads in output file to a JS object.
+const runTest = function(benchmarksCsv) {
+  const cmd = 'cat ' + testCsv + ' | ' + 'node ./scripts/benchmarks/parse-benchmarks-data.js 2015 2017';
+  const benchmark = execSync(cmd).toString();
+  return JSON.parse(benchmark);
+};
+
+describe('parse benchmark data', function() {
+  it('should create new measures and ignore duplicate measureIds', () => {
```
Comment (kencheeto) [inline test/scripts/benchmarks/parse-benchmark-data-spec.js]: if we throw an error when a fuzzy matched measure id can't be found, could we reduce this test case down to checking for duplicate measureIds (using a smaller test csv, checking for specific measure ids in output)? this is a hard test case to maintain; if it breaks i'm not sure how I'd debug what went wrong.

File: test/scripts/benchmarks/parse-benchmark-data-spec.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    const measures = runTest(testCsv);
```
Comment (kencheeto) [inline test/scripts/benchmarks/parse-benchmark-data-spec.js]: would `benchmarks` be a better variable name?

**Pull request merged**